### PR TITLE
CBG-587: Code coverage for rest package (api.go)

### PR DESCRIPTION
**Pull request by sarathkumarsivan**
- Add coverage for handleProfiling

### CODE REVIEW

File: rest/api_test.go [hunk 1]
```diff
@@ -4466,3 +4467,316 @@ func Benchmark_RestApiGetDocPerformanceFullRevCache(b *testing.B) {
...
+		{
+			profileName:                 "goroutine",
+			shortDescription:            "Get request for pprof goroutine",
+			inputMethod:                 http.MethodGet,
+			inputResource:               "/_debug/pprof/goroutine?debug=0&gc=1",
+			inputReqBody:                "",
+			expectedContentDisposition:  `attachment; filename="goroutine"`,
+			expectedContentType:         "application/octet-stream",
+			expectedXContentTypeOptions: "nosniff",
+			expectedStatus:              http.StatusOK},
```
Comment (adamcfraser) [inline rest/api_test.go]: There are several values that you expect to be the same for every test (inputReqBody, expectedXContentTypeOptions, etc).  Why are they all being defined in the test struct?

**Review by adamcfraser — CHANGES_REQUESTED** [review]

File: rest/api_test.go (hunk 1 above)
Comment (sarathkumarsivan) [reply rest/api_test.go]: This is fixed and the test has been rewritten.

File: rest/api_test.go [hunk 2]
```diff
@@ -4466,3 +4467,205 @@ func Benchmark_RestApiGetDocPerformanceFullRevCache(b *testing.B) {
...
+		{inputProfile: "threadcreate"},
+		{inputProfile: "heap"},
+		{inputProfile: "allocs"},
+		{inputProfile: "block"},
+		{inputProfile: "mutex"},
+	}
+
+	for _, tc := range tests {
+		// Send a valid profile request.
+		resource := fmt.Sprintf("/_profile/%v", tc.inputProfile)
```
Comment (bbrks) [inline rest/api_test.go:4492]: All of these tests calling the profiling endpoints take a decent chunk of time each - maybe we should skip these when running in short mode, or evaluate whether we actually need to test these at all.

**Review by bbrks — CHANGES_REQUESTED** [review]

File: rest/api_test.go [hunk 3]
```diff
@@ -4466,3 +4467,205 @@ func Benchmark_RestApiGetDocPerformanceFullRevCache(b *testing.B) {
 		}
 	})
 }
+
+func TestHandleProfiling(t *testing.T) {
+	defer base.SetUpTestLogging(base.LevelDebug, base.KeyAll)()
```
[the remark this answers is not in the document]
Comment (sarathkumarsivan) [reply rest/api_test.go]: Makes sense; this is not really required. Let's remove it from all the profiling related tests. Changes are available in the latest revision.

File: rest/api_test.go (hunk 2 above)
Comment (sarathkumarsivan) [reply rest/api_test.go:4492]: Yeah, it takes close to 12 seconds during each run; let's skip these when running in short mode.

**Review by adamcfraser — APPROVED** [review]

**Pull request merged**